console/src/warnings.rs

```diff
@@ -0,0 +1,37 @@
+use crate::tasks::Task;
+
+pub trait Warning<T>: std::fmt::Debug {
```

Comment:
This seems fine to start with. Maybe we discover we need something different when trying to add new warnings, but no point in trying to imagine.

Reply:
yeah, i figured the trait can be pretty minimal to start and we can add stuff as we add more warnings. one thing i'd really like to have in the eventual future is some way of generating structured web-based documentation for the different categories of warnings/lints, explaining why they might be a problem, how to fix them, etc, like [what clippy has](https://rust-lang.github.io/rust-clippy/master/index.html#). then, we can have a URL and/or an lint number users can use to get more information. but, i think adding all of that infrastructure is a fairly big project...perhaps we can crib some of it from clippy?

similarly, it could eventually be nice to have categories of warnings/lints, and allow users to turn some of them on or off. but that's not important unless we have a _big_ list of warnings.